src/backend/looped/vehicle/navigation.cpp

```diff
@@ -0,0 +1,256 @@
+#include "backend/looped_command.hpp"
+#include "fiber_pool.hpp"
+#include "gta/enums.hpp"
+#include "natives.hpp"
+#include "pointers.hpp"
+#include "script.hpp"
+#include "util/audio.hpp"
+#include "util/blip.hpp"
+#include "util/entity.hpp"
+#include "util/math.hpp"
+#include "util/notify.hpp"
+#include "util/vehicle.hpp"
+
+
+//70 feet annouce the next action plus what is coming up
+
+
+namespace big
+{
+	class navigation : looped_command
+	{
+		//bool needsNextDirection = true;
+		using looped_command::looped_command;
+
+		virtual void on_enable() override
+		{
+			g_notification_service->push_success("Navigation", "Navigation has been enabled");
+		}
+
+		virtual void on_tick() override
+		{
+			g_fiber_pool->queue_job([] {
+				int direction;
+				int feetToAction;
+				Vector3 waypoint;
+				float dist;
+				float v;
+
+				blip::get_blip_location(waypoint, (int)BlipIcons::Waypoint);
+				bool doesPointExist = blip::get_blip_location(waypoint, (int)BlipIcons::Waypoint);
+
+				if (doesPointExist)
+				{
+					PATHFIND::GENERATE_DIRECTIONS_TO_COORD(waypoint.x, waypoint.y, waypoint.z, 1, &direction, &v, &dist);
+					feetToAction = math::divided_by_ten(math::meters_to_feet(dist));
+					switch (direction)
+					{
+					case 1:
+						if (g.self.nav_text)
```

Comment:
Just make a helper function that will check both `nav_text` or `nav_voice` why do it every time?

Reply:
so just do like if(checkvoice()) or something?